diff --git a/external/DifferentialEq.cpp b/external/DifferentialEq.cpp
index 6a66277..4e2396f 100644
--- a/external/DifferentialEq.cpp
+++ b/external/DifferentialEq.cpp
@@ -88,6 +88,7 @@ SYCL_EXTERNAL void differentialEq(
         Actor neighbour = actors[x];
 
         bool bBoxFlag{false};
+        #pragma unroll
         for (unsigned int iBox{0}; iBox<9; ++iBox) {
             bBoxFlag |= (neighbourBoxes[iBox][0]==neighbour.getBBox()[0] &&
                          neighbourBoxes[iBox][1]==neighbour.getBBox()[1]);
